### ISSUE TITLE
Question on implementation of classification subnet

### ISSUE THREAD

**Issue by ruizengalways** [fizyr/keras-retinanet#522]
I notice that COCO dataset has 80 classes. So there should be 81 (80 classes + 1 background class) neurons in the last layer of the classfication subnet. But the implementation only has only 80 neurons. The author seems implicitly encode background class vector as a zero vector rather than a one-hot vector like other normal classes. So what If I modify it to 81 neurons and still use the same loss function, i.e. the background class is explicitly encoded as a one-hot vector which the 81th element is 1. Is it still correct to do this regarding mathematic? Could anyone explain this to help me? Thank you.

**Comment by yhenon** [comment]
>I notice that COCO dataset has 80 classes. So there should be 81 (80 classes + 1 background class) neurons in the last layer of the classfication subnet. But the implementation only has only 80 neurons. The author seems implicitly encode background class vector as a zero vector rather than a one-hot vector like other normal classes.

The current implementation is correct and matches the paper. It uses sigmoid activations + binary cross-entropy loss rather than softmax activation + categorical crossentropy.

>So what If I modify it to 81 neurons and still use the same loss function, i.e. the background class is explicitly encoded as a one-hot vector which the 81th element is 1. Is it still correct to do this regarding mathematic? Could anyone explain this to help me? Thank you.

You would need to modify the loss function and the initialization scheme accordingly.

Closing this as it not an issue with the codebase.
